drivers/web-worker-driver/src/main/kotlin/app/cash/sqldelight/driver/worker/api/WorkerRequest.kt

```diff
@@ -0,0 +1,34 @@
+package app.cash.sqldelight.driver.worker.api
+
+/**
+ * Messages sent by the SQLDelight driver to the worker.
+ */
+@OptIn(ExperimentalJsExport::class)
+@JsExport
+external interface WorkerRequest {
+  /**
+   * A unique identifier used to identify responses to this message
+   * @see WorkerResponse.id
+   */
+  var id: Int
+
+  /**
+   * The action that the worker should run. Can be one of the following values:
+   *
+   * - `exec`: Execute the given [sql] with the given [params]
+   * - `begin_transaction`: Begin a transaction
+   * - `end_transaction`: End/commit a transaction
+   * - `rollback_transaction`: Rollback the current transaction
+   */
+  var action: String
```

Comment:
You could also use some hacks to use an enum: https://github.com/JetBrains/compose-jb/blob/master/web/core/src/jsMain/kotlin/org/jetbrains/compose/web/css/CSSUnits.kt

Reply:
I'm having some trouble understanding what all that code is doing. I'm especially startled by `@JsName("""(/*union*/{auto: 'auto'}/*union*/)""")` how does that even work 😳 

Is it that the enum will be stringified into the string equivalent when `JSON.stringify`-ed?

Reply:
Not really, during js code generating this "name" is used directly in the js code without any kotlin overhead.

Reply:
Ok I think I understand. I'll give it a shot and see if I can get it to work.

Reply:
This sample might be simpler to understand: https://github.com/JetBrains/compose-jb/blob/5d18d0a79dd26a0d5880bdb8c0bb7b8870c9421b/web/core/src/jsMain/kotlin/org/jetbrains/compose/web/attributes/PredefinedAttrValues.kt#L213

Reply:
That makes sense! Thanks